commands/command_ls_files.go

```diff
@@ -105,8 +105,16 @@ func lsFilesCommand(cmd *cobra.Command, args []string) {
 	includeArg, excludeArg := getIncludeExcludeArgs(cmd)
 	gitscanner.Filter = buildFilepathFilter(cfg, includeArg, excludeArg)
 
-	if err := gitscanner.ScanIndex(ref.Sha, nil); err != nil {
-		Exit("Could not scan for Git LFS index: %s", err)
+	if currentRef.Sha == ref.Sha {
+		// Only scan the index when "git lfs ls-files" was invoked with
+		// no arguments, or a reference pointing to the HEAD checkout
+		// (or before the first commit).
```

Comment:
Do we still want to scan the index even if the user has specified an explicit reference pointing to HEAD?  I would think that we'd only want to list the index if no treeish was given.

Consider the case that we have a branch also pointing to the same commit as this branch, or the case of a scripting tool which has provided some arbitrary resolved ref which happens to resolve to HEAD (e.g. a tag).